docker/snippets/rosdep.dockerfile

```diff
@@ -12,16 +12,29 @@ RUN apt-get update && \
 ARG ROS_DISTRO
 ARG ROS_WORKSPACE="./workspace"
 ARG ROSDEP_METAPACKAGE=""
+ARG ROSDEP_OS_NAME="ubuntu"
+ARG ROSDEP_OS_VERSION="jammy"
+ARG ROSDEP_SKIP_KEYS=""
 ARG ROS_INSTALL_PREFIX="/opt/ros/${ROS_DISTRO}/"
 COPY ${ROS_WORKSPACE}/src /tmp/workspace/src
 RUN . ${ROS_INSTALL_PREFIX}/setup.sh && \
     cd /tmp/workspace && \
     apt-get update && \
-    if [ -z "$ROSDEP_METAPACKAGE" ]; then \
+    if [[ -z "${ROSDEP_METAPACKAGE}" ]]; then \
         ROSDEP_FROM_PATHS="src"; \
     else \
-        ROSDEP_FROM_PATHS=$(colcon list --packages-up-to $ROSDEP_METAPACKAGE | awk '{print $2}' | tr '\n' ' '); \
+        ROSDEP_FROM_PATHS=$(colcon list --packages-up-to ${ROSDEP_METAPACKAGE} | awk '{print $2}' | tr '\n' ' '); \
     fi && \
-    rosdep install --from-paths $ROSDEP_FROM_PATHS --ignore-src -r -y && \
+    rosdep install --from-paths ${ROSDEP_FROM_PATHS} --ignore-src -r -y --skip-keys="${ROSDEP_SKIP_KEYS}" --os=${ROSDEP_OS_NAME}:${ROSDEP_OS_VERSION} --rosdistro=${ROS_DISTRO} && \
     rm -rf /tmp/workspace && \
     apt-get clean && apt-get autoremove -y && rm -rf /var/lib/apt/lists/*
+
+# Avoid setup.py and easy_install deprecation warnings caused by colcon and setuptools
+# https://github.com/colcon/colcon-core/issues/454
+ENV PYTHONWARNINGS=ignore:::setuptools.command.install,ignore:::setuptools.command.easy_install,ignore:::pkg_resources
+RUN echo "Warning: Using the PYTHONWARNINGS environment variable to silence setup.py and easy_install deprecation warnings caused by colcon"
+
+# Fix permissions since we installed everything as root
+USER ${USER}
+RUN rosdep fix-permissions
```

Comment:
getting permission issues... 
![image](https://github.com/uwsbel/autonomy-research-testbed/assets/52421529/cf6275a2-ceb5-4f2b-a59c-d5af7ddeb633)

Reply:
What computer is this on? It looks like you built the `dev` service, not `art5`? Just remove the log/build/install directory using sudo and try again. You can see what your user is named using `whoami`, but it will be art since that's what the `art@art-dev` means. Maybe try checking `id -u` and `id -g` to see if you set those correctly because it looks like this is on your workstation, not a jetson.

Reply:
I don't see permission issues on my computer or the jetson.